private/syntax/compile/binding-spec.rkt

```diff
@@ -53,16 +55,20 @@
 (struct ref [pvar] #:transparent)
 (struct bind with-stx [pvar] #:transparent)
 (struct bind-syntax with-stx [pvar transformer-pvar] #:transparent)
-(struct bind-syntaxes with-stx [pvar transformer-pvar] #:transparent)
-(struct rec with-stx [pvars] #:transparent)
-(struct re-export with-stx [pvars] #:transparent)
+(struct bind-syntaxes with-stx [depth pvar transformer-pvar] #:transparent)
+(struct rec with-stx [depth pvar] #:transparent)
```

Comment:
How about we call `rec` `import`? And it shouldn't need a depth, because we will expect ellipses outside it only.